### ISSUE TITLE
please add copyright statement

### ISSUE THREAD

**Issue by onlyjob** [RoaringBitmap/roaring#62]
Perhaps the best place to document copyright (by adding something like `Copyright 2016 owner <email@host.org>`) would be `README.md`. Thanks.

**Comment by lemire** [comment]
Why?

**Comment by onlyjob** [comment]
Because it is missing. It is unclear who grants the license.
_roaring_ may be introduced to [Debian](http://debian.org) but initial compliance review was unsuccessful due to lack of copyright statement.
Documenting copyrights is a usual (and necessary) practice.

**Comment by onlyjob** [comment]
That's better than nothing but still not ideal as "the contributors" is not a legal entity.

Perhaps main contributor (or organisation) can be mentioned in copyright statement?

To me "the contributors" suggests that legal status of the project (ownership) may be uncertain...

Also word "by" is not necessary in copyright statements... Thank you.

**Comment by statik** [comment]
I don't think it is a useful thing to put go libraries in debian packaging. What is the thing that would be installed by the debian package? Go libraries don't get distributed as shared libraries, they get distributed as source, and the consumer of a go library compiles everything into a static binary.

**Comment by statik** [comment]
Aha, I found this page http://pkg-go.alioth.debian.org/packaging.html which explains that library only packages are not available to end users, they are only for use by debian packaging system when building binaries written in go. So in this case I understand why debian would like to create source packages for go libraries.

**Comment by onlyjob** [comment]
Go libraries are packaged as `-dev` packages to satisfy build dependencies. Those packages are only shipping sources but their build tests are running on all Golang architectures -- a something that is so rarely done for vendored libraries.

**Comment by lemire** [comment]
@onlyjob 

The copyright of the Go language itself is given as : "Copyright (c) 2012 The Go Authors."  See : https://golang.org/LICENSE

Is Go included in debian? 

The linux kernel is copyrighted by "Linus Torvalds and others who actually wrote it".

Presumably, the Linux kernel is part of Debian?

Git's copyright is "owned" by all sorts of people including Linus Torvalds, me (of all people) and so forth... https://github.com/git/git 

Surely, git is part of Debian?

If you prefer, we can rephrase is as "Copyright the roaring Authors".


Or if you prefer we can do like we did with JavaEWAH: https://github.com/lemire/javaewah

> (c) 2009-2016 Daniel Lemire (http://lemire.me/en/), Cliff Moon, David McIntosh (https://github.com/mctofu), Robert Becho (https://github.com/RBecho), Colby Ranger (https://github.com/crangeratgoogle), Veronika Zenz (https://github.com/veronikazenz), Owen Kaser (https://github.com/owenkaser), Gregory Ssi-Yan-Kai (https://github.com/gssiyankai), and Rory Graves (https://github.com/rorygraves)

That'd be fine.

Note that javaewah is part of debian already so it ought to be good enough.

Please let us know what you think.

**Comment by lemire** [comment]
@onlyjob  Note that I'd genuinely want to clear this up with you.

**Comment by lemire** [comment]
@statik  I would definitively be a positive to be included by debian.

If all that is blocking us is a clearer copyright notice. Let us solve that.

**Comment by onlyjob** [comment]
I don't have time for analysis of copyright documentation in the mentioned projects but usually projects that use _copyright "the contributors"_ also maintain list of contributors (copyright holders)
or headers in source files with details about license and copyright holders specific to a particular file.
Sometimes there are two (different) lists -- `CONTRIBUTORS` (maintained for copyright purposes) and `AUTHORS`.

Here unfortunately there is no copyright information whatsoever so _"Copyright 2016 by the contributors"_ makes no sense (what contributors?), much like notoriously meaningless [All rights reserved](https://en.wikipedia.org/wiki/All_rights_reserved) statement.

Some projects use something like "Copyright 2015 CoreOS Inc. and contributors" but again mentioning contributors is meaningful only when at least some of them are listed somewhere.

I suggest to identify main contributor(s) or organisation(s) who chose a license for a project and list them as copyright holders.

_JavaEWAH_ copyright documentation is OK but it is not properly formatted: it should be list with years range and one contributor per line, like the following

~~~~
Copyright 2010-2016 Cliff Moon
Copyright 2009-2016 Daniel Lemire (http://lemire.me/en/)
Copyright 2011-2016 David McIntosh (https://github.com/mctofu)
~~~~

(I've made up copyright years to illustrate the idea).

**Comment by lemire** [comment]
@onlyjob 

> use copyright "the contributors" also maintain list of contributors (copyright holders)
or headers in source files with details about license and copyright holders specific to a particular file.
Sometimes there are two (different) lists -- CONTRIBUTORS (maintained for copyright purposes) and AUTHORS.

Ok.

See https://github.com/RoaringBitmap/roaring/commit/e9cfdabbfbe11865b64a7658d83c1b6fc3bf92ec

**Comment by onlyjob** [comment]
That'll do, thank you. :)
Though I like JavaEWAH approach is better because it is just a straightforward list of copyright holders in one place. Here we have copyright statement in README and list of contributors in another file, de-coupled from copyright years. Wouldn't it be simpler to maintain all this information in README?

**Comment by lemire** [comment]
@onlyjob 

We currently have exactly the same setup as the golang project itself.

**Comment by onlyjob** [comment]
Fair enough. :) But I'm not convinced that Golang is an example of best practice in regards to copyright documentation...
